src/BuildScriptGenerator/PlatformInstallerBase.cs

```diff
@@ -0,0 +1,97 @@
+﻿// --------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+// --------------------------------------------------------------------------------------------
+
+using System;
+using System.Linq;
+using System.Text;
+using Microsoft.Extensions.Options;
+using Microsoft.Oryx.Common;
+
+namespace Microsoft.Oryx.BuildScriptGenerator
+{
+    public abstract class PlatformInstallerBase
+    {
+        protected readonly BuildScriptGeneratorOptions _commonOptions;
+        protected readonly IEnvironment _environment;
+
+        public PlatformInstallerBase(IOptions<BuildScriptGeneratorOptions> commonOptions, IEnvironment environment)
+        {
+            _commonOptions = commonOptions.Value;
+            _environment = environment;
+        }
+
+        public abstract string GetInstallerScriptSnippet(string version);
+
+        public abstract bool IsVersionAlreadyInstalled(string version);
+
+        protected string GetInstallerScriptSnippet(
+            string platformName,
+            string version,
+            string directoryToInstall = null)
+        {
+            var sdkStorageBaseUrl = GetPlatformBinariesStorageBaseUrl();
+
+            var versionDirInTemp = directoryToInstall;
+            if (string.IsNullOrEmpty(versionDirInTemp))
+            {
+                versionDirInTemp = $"{Constants.TempInstallationDirRoot}/{platformName}/{version}";
+            }
+
+            var tarFile = $"{version}.tar.gz";
+            var snippet = new StringBuilder();
+            snippet
+                .AppendLine()
+                .AppendLine("PLATFORM_SETUP_START=$SECONDS")
+                .AppendLine("echo")
+                .AppendLine($"echo Downloading and installing {platformName} version '{version}'...")
+                .AppendLine($"rm -rf {versionDirInTemp}")
+                .AppendLine($"mkdir -p {versionDirInTemp}")
+                .AppendLine($"cd {versionDirInTemp}")
+                .AppendLine(
```

Comment:
nit: is it possible to add some timer before and after curl to measure the download time?